libs/pika/async_mpi/tests/unit/algorithm_transform_mpi.cpp

```diff
@@ -114,10 +112,9 @@ int pika_main()
                 // MPI function pointer
                 int data = 0, count = 1;
                 if (rank == 0) { data = 42; }
-                auto result = tt::sync_wait(
+                tt::sync_wait(
```

Comment:
Do I understand correctly that functions that return `MPI_Status` are now special cased to turn that into an exception (if not success)? What about functions that don't return `MPI_Status` (nor void)?

Reply:
we only handle mpi type functions that return mpi status and we set any error on the set_error. I can put back the arbitrary return types if necessary, but I'd prefer to do that in a PR when it is needed by something. Removing it simplified my work a bit initially.